classes/userGroup/UserGroup.php

```diff
@@ -70,11 +70,11 @@ public function getContextId()
     /**
      * Set the context ID
      *
-     * @param int $contextId
+     * @param ?int $contextId
```

Comment:
Since we're early in the 3.4.0 dev cycle, I'd suggest just making this a first-class type hint. (Also elsewhere)